terraform/modules/tf_stream_alert_athena/main.tf

```diff
@@ -0,0 +1,137 @@
+// IAM Role: Lambda Execution Role
+resource "aws_iam_role" "athena_partition_role" {
+  name = "streamalert_athena_partition_refresh"
+
+  assume_role_policy = "${data.aws_iam_policy_document.lambda_assume_role_policy.json}"
+}
+
+// IAM Policy Doc: Generic Lambda trust relationship policy
+data "aws_iam_policy_document" "lambda_assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+// IAM Role Policy: Allow the Lambda function to execute Athena queries
+// Ref: http://amzn.to/2vJqUAA
+resource "aws_iam_role_policy" "athena_query_permissions" {
+  name = "streamalert_athena_partition_refresh"
+  role = "${aws_iam_role.athena_partition_role.id}"
+
+  policy = "${data.aws_iam_policy_document.athena_query_policy.json}"
+}
+
+// IAM Policy Doc: Athena, CloudWatch, and S3 permissions
+data "aws_iam_policy_document" "athena_query_policy" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+
+    resources = [
+      "arn:aws:logs:*:*:*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "athena:RunQuery",
+      "athena:GetQueryExecution",
+      "athena:GetQueryExecutions",
+      "athena:GetQueryResults",
+      "athena:StartQueryExecution",
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetBucketLocation",
+      "s3:GetObject",
+      "s3:ListBucket",
+      "s3:ListBucketMultipartUploads",
+      "s3:ListMultipartUploadParts",
+      "s3:PutObject",
+    ]
+
+    resources = [
+      "arn:aws:s3:::aws-athena-query-results-*",
```

Comment:
Just want to confirm that the resources has hyphen sign after `results` (``).
[AWS docs](http://docs.aws.amazon.com/athena/latest/ug/access.html) shows it wildcards everything after results which has wider range
```
"Resource": [
        "arn:aws:s3:::aws-athena-query-results*"
      ]
```